src/index.ts

```diff
@@ -564,6 +573,63 @@ class threads {
     });
   };
 
+  @muxrpc('async')
+  public recentHashtags = (opts: RecentHashtagsOpts, cb: CB<Array<string>>) => {
+    if (typeof opts.limit !== 'number' || opts.limit <= 0)
+      return cb(new Error('Limit must be number greater than 0'));
+
+    const preserveCase = !!opts.preserveCase;
+
+    // completely normalized hashtag (no leading # and lowercase) -> partially normalized (no leading #)
+    const result = new Map<string, string>();
+
+    return pull(
+      this.ssb.db.query(
+        where(and(isPublic(), hasSomeHashtag())),
+        descending(),
+        batch(BATCH_SIZE),
+        toPullStream(),
+      ),
+      this.removeMessagesFromBlocked,
+      pull.through((msg: Msg<any>) => {
+        const { channel, mentions } = msg.value.content;
+
+        if (channel) {
+          const withoutHashtag = withoutLeadingHashtag(channel);
+          const lowercaseWithoutHashtag = withoutHashtag.toLocaleLowerCase();
+
+          // Since the messages received are descending,
+          // we don't want to update the value for associated key if it already exists
+          // because we want the keep the most recent variation of the hashtag (accounts for casing)
+          if (!result.has(lowercaseWithoutHashtag)) {
+            result.set(lowercaseWithoutHashtag, withoutHashtag);
+          }
+        } else if (Array.isArray(mentions)) {
+          // Assumes `mentions` goes from oldest to most recent
+          // so we want to process them in reverse order
+          for (const mention of mentions.reverse()) {
+            const withoutHashtag = withoutLeadingHashtag(mention.link);
```

Comment:
do we need a runtime check to make sure `mention.link` exists here, or will the db2 query that we get the message from take care of that for us?

Reply:
We don't need the runtime check, because the `processRecord` method in the HashtagsPlugin checks that, and since the operator `hasSomeHashtag()` comes from HashtagsPlugin, we know that all the msgs we are processing here have already been processed by HashtagsPlugin. See the part in `processRecord` that plucks the `link` from `mention`.